src/Modules/Workspaces/Commands.Workspaces/GetPowerBIWorkspace.cs

```diff
@@ -165,8 +163,34 @@ private void ExecuteCmdletWithAll()
                         .ToList();
                 }
 
-                this.Logger.WriteObject(allWorkspaces, true);
+                if (this.Deleted.IsPresent)
+                {
+                    filteredWorkspaces.AddRange(allWorkspaces.Where(w => w.State.Equals(WorkspaceState.Deleted)));
+                }
+
+                if (this.Orphaned.IsPresent)
+                {
+                    filteredWorkspaces.AddRange(allWorkspaces.Where(w => this.IsOrphanWorkspace(w)));
+                }
+                
+                if(!this.Deleted.IsPresent && !this.Orphaned.IsPresent)
+                {
+                    this.Logger.WriteObject(allWorkspaces, true);
+                    return;
+                }
+
+                this.Logger.WriteObject(filteredWorkspaces, true);
             }
         }
+
+        private bool IsOrphanWorkspace(Workspace w)
+        {
+            if (w.State.Equals(WorkspaceState.Deleted))
+            {
+                return false;
+            }
+
+            return (w.Users == null || w.Users.Count() == 0) || (!w.Users.Any(u => u.AccessRight.Equals(WorkspaceUserAccessRight.Admin.ToString())));
```

Comment:
Make sure that it will not return group based workspaces as groups does not return users even when it is not an orphaned workspace.